extensions/amp-fx-collection/0.1/providers/amp-fx-presets.js

```diff
@@ -84,36 +75,24 @@ export const Presets = {
         return;
       }
 
-      if (fxElement.isMutateScheduled()) {
-        return;
-      }
-
       // only do this on the first element
       if (!fxElement.initialTrigger) {
-        fxElement.getResources().mutateElement(
-            fxElement.getElement(), function() {
-              const style = computedStyle(fxElement.getAmpDoc().win,
-                  fxElement.getElement());
-              setStyles(fxElement.getElement(), {
-                'top': `calc(${style.top} + ${fxElement.getFlyInDistance()}vh)`,
-                'visibility': 'visible',
-              });
-              fxElement.initialTrigger = true;
-            });
+        const style = computedStyle(fxElement.getAmpDoc().win,
```

Comment:
@nainar with this change, we are assuming the element is laidout by the time the `fx` triggers which may or ma not be the case. I think we need to keep the mutateElement for the initial trigger.
a good test for this would be fly-in-left for an image that more than 3 viewport height down the page  initially.

Reply:
Made the change in the latest PR you are right. Was broken for one of the examples in `article-fly-in.amp.html`